addons/web_editor/static/src/scss/wysiwyg_snippets.scss

```diff
@@ -245,6 +245,24 @@ body.editor_enable.editor_has_snippets {
             text-transform: capitalize;
         }
 
+        we-imagepicker {
+            we-row {
+                max-width: 100%;
+            }
+
+            .o_we_pick_image {
+                display: flex;
+                justify-content: space-between;
+                align-items: center;
+                & > span {
+                    // 1.2 em is the width of fontawesome icons
+                    max-width: calc(100% - 1.2em);
+                    overflow: hidden;
+                    text-overflow: ellipsis;
+                }
+            }
+        }
```

Comment:
Each time you use calc, I lose 1 year of my life and a kitten dies :( I guess this scss will be reviewed with new widget structure anyway ;)

Reply:
The worst part is it's not even needed. When I was fiddling with this I had the problem that the text was pushing the edit icon outside of the button, even with overflow hidden and text-overflow ellipsis.